### ISSUE TITLE
Allow snmptrap to read USM user from persistent snmpd.conf?

### ISSUE THREAD

**Issue by batkins61** [net-snmp/net-snmp#32]
It would be great if snmptrap could have an option to take USM credentials from a user configured in the persistent snmpd.conf file, similar to how trapsess does for snmpd.  That way only the user, and not the passphrases or keys would need to be included on the command line.

I have a AgentX sub-agent that needs to send V3 traps/informs, and currently I'm parsing the persistent snmpd.conf file for the users and localized keys and then using them to send the traps to avoid storing and using passwords on the command line, which seems like something snmptrap could so with -u <user> and another option to enable the behavior.

**Comment by hardaker** [comment]
It would be insecure to let a random user on the system access snmpd.conf files.  The right way to do this is with a generic snmp.conf file that sets the correct authentication information.  See the SNMPv3 SETTINGS section in the snmp.conf manual page.

**Comment by batkins61** [comment]
Of course, the persistent snmpd.conf file, where the localized passphrases are, would have restricted permissions, and the snmptrapd process would have to have permission to read that file to utilize that feature. That's hardly sufficient reason to mark the request invalid.

**Comment by hardaker** [comment]
Sorry; the invalid tag is not exactly a friendly word.  But let's try this instead: what are your underlying requirements.  I *think* I may now understand that you're really just talking about your agentx subagent and not the snmptrap CLI command.   Or maybe you're calling the snmptrap command from your agentx subagent?  If so, why is this being done instead of sending the trap via the agentx channel per normal and having the agent send out the notifications to its registered set of receivers (which is how agentx was designed).  I'm trying to understand the underlying goals that are causing you to do something different than normal (which I'm certainly not implying as wrong; I'm just trying to get a clear picture of how you're using things).
